File: [404] 左叶子之和.cpp
#include <cstdlib>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
	int sumOfLeftLeaves(TreeNode* root) {
		if (root == nullptr || (root->left == nullptr && root->right == nullptr))
			return 0;
		
		return (root->left && root->left->left == nullptr && root->left->right == nullptr ?
			root->left->val : 0) + sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right);
	}
};

File: [404]左叶子之和.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        return self.dp(root, False)

    def dp(self, root, is_left):
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return root.val if is_left else 0
        else:
            return self.dp(root.left, True) + self.dp(root.right, False)
